sdk-lib/src/main/rust/lib.rs

```diff
@@ -292,22 +285,20 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_tool_DerivationTool_deriveU
     let res = panic::catch_unwind(|| {
         let network = parse_network(network_id as u32)?;
         let ufvk_string = utils::java_string_to_rust(&env, ufvk_string);
-        let ufvk = match utils::fake_ufvk_decode(&ufvk_string) {
-            Some(ufvk) => ufvk,
-            None => {
+        let ufvk = match UnifiedFullViewingKey::decode(&network, &ufvk_string) {
+            Ok(ufvk) => ufvk,
+            Err(e) => {
                 return Err(format_err!(
-                    "Error while deriving viewing key from string input"
+                    "Error while deriving viewing key from string input: {}",
+                    e,
                 ));
             }
         };
 
-        // Derive the default Sapling payment address (like older SDKs used).
-        let (di, sapling) = ufvk.1.default_address().unwrap();
-        // Derive the transparent address corresponding to the default Sapling diversifier
-        // index (matching ZIP 316).
-        let p2pkh = utils::p2pkh_addr(ufvk.0, di).unwrap();
-        // TODO: replace this with `zcash_address::unified::Address`.
-        let address_str = utils::fake_ua_encode(&p2pkh, &sapling);
+        // Derive the default Unified Address (containing the default Sapling payment
+        // address that older SDKs used).
+        let (ua, _) = ufvk.default_address();
```

Comment:
This reminds me that we need to ensure that we handle search for the child 0 "diversified" transparent address correctly even when the Sapling default address is at a different diversifier index.

Reply:
Please file an issue.

Reply:
https://github.com/zcash/librustzcash/issues/599